scripts/serve_header/serve_header.py

```diff
@@ -0,0 +1,381 @@
+        try:
+            super().__init__(request, client_address, server,
+                             directory=server.worktrees.root_dir)
+        except ConnectionResetError:
+            logging.debug('connection reset by peer')
+
+    def translate_path(self, path):
+        path = os.path.abspath(super().translate_path(path))
+
+        # add single_include/nlohmann into path, if needed
+        header = os.path.join('/', HEADER)
+        header_path = os.path.join('/', SINGLE_INCLUDE, HEADER)
+        if (path.endswith(header)
+            and not path.endswith(header_path)):
+            path = os.path.join(os.path.dirname(path), SINGLE_INCLUDE, HEADER)
+
+        return path
+
+    def send_head(self):
+        # check if the translated path matches a working tree
+        # and fullfill the request; otherwise, send 404
+        path = self.translate_path(self.path)
+        self.worktree = self.worktrees.find(path)
+        if self.worktree is not None:
+            self.worktree.amalgamate_header()
+            logging.info(f'{self.worktree.name}; serving header (build count {self.worktree.build_count})')
+            return super().send_head()
+        logging.info(f'invalid request path: {self.path}')
+        super().send_error(HTTPStatus.NOT_FOUND, 'Not Found')
+        return None
+
+    def send_header(self, keyword, value):
+        # intercept Content-Length header; sent in copyfile later
+        if keyword == 'Content-Length':
+            return
+        super().send_header(keyword, value)
+
+    def end_headers (self):
+        # intercept; called in copyfile or indirectly
+        # by send_head via super().send_error
+        pass
+
+    def copyfile(self, source, outputfile):
+        injected = False
+        content = BytesIO()
+        length = 0
+        # inject build count and time into served header
+        for line in source:
+            line = line.decode('utf-8')
+            if not injected and JSON_VERSION_RE.match(line):
+                length += content.write(bytes('#define JSON_BUILD_COUNT '\
+                                              f'{self.worktree.build_count}\n', 'utf-8'))
+                length += content.write(bytes('#define JSON_BUILD_TIME '\
+                                              f'"{self.worktree.build_time}"\n\n', 'utf-8'))
+                injected = True
+            length += content.write(bytes(line, 'utf-8'))
+
+        # set content length
+        super().send_header('Content-Length', length)
+        # CORS header
+        self.send_header('Access-Control-Allow-Origin', '*')
+        # prevent caching
+        self.send_header("Cache-Control", "no-cache, no-store, must-revalidate")
+        self.send_header("Pragma", "no-cache")
+        self.send_header("Expires", "0")
+        super().end_headers()
+
+        # send the header
+        content.seek(0)
+        shutil.copyfileobj(content, outputfile)
+
+    def log_message(self, format, *args):
```

Comment:
If I change it to `fmt`, Codacy complains the parameter name doesn't match the one of the overridden method. ;-)

Reply:
You'll have to silence the other ones anyone but I thought I should point this one out.

Reply:
Alright.

Reply:
The issues should be ignored in the next build.